.test-infra/jenkins/common_job_properties.groovy

```diff
@@ -273,4 +259,107 @@ class common_job_properties {
         shell("python PerfKitBenchmarker/pkb.py $pkbArgs")
     }
   }
+
+  /**
+   * Sets properties for all jobs which are run by a pipeline top-level (maven) job.
+   * @param context    The delegate from the top level of a MavenJob.
+   * @param jobTimeout How long (in minutes) to wait for the job to finish.
+   * @param descriptor A short string identifying the job, e.g. "Java Unit Test".
+   */
+  static def setPipelineJobProperties(def context, int jobTimeout, String descriptor) {
+    context.parameters {
+      stringParam(
+              'ghprbGhRepository',
+              'N/A',
+              'Repository name for use by ghprb plugin.')
+      stringParam(
+              'ghprbActualCommit',
+              'N/A',
+              'Commit ID for use by ghprb plugin.')
+      stringParam(
+              'ghprbPullId',
+              'N/A',
+              'PR # for use by ghprb plugin.')
+
+    }
+
+    // Set JDK version.
+    jdk('JDK 1.8 (latest)')
+
+    // Restrict this project to run only on Jenkins executors as specified
+    label('beam')
+
+    // Execute concurrent builds if necessary.
+    concurrentBuild()
+
+    wrappers {
+      timeout {
+        absolute(jobTimeout)
+        abortBuild()
+      }
+      credentialsBinding {
+        string("COVERALLS_REPO_TOKEN", "beam-coveralls-token")
+      }
+      downstreamCommitStatus {
+        context("Jenkins: ${descriptor}")
+        triggeredStatus("${descriptor} Pending")
+        startedStatus("Running ${descriptor}")
+        statusUrl()
+        completedStatus('SUCCESS', "${descriptor} Passed")
+        completedStatus('FAILURE', "${descriptor} Failed")
+        completedStatus('ERROR', "Error Executing ${descriptor}")
+      }
+      // Set SPARK_LOCAL_IP for spark tests.
+      environmentVariables {
+        env('SPARK_LOCAL_IP', '127.0.0.1')
+      }
+    }
+
+    // Set Maven parameters.
+    setMavenConfig(context)
+  }
+
+  /**
+   * Sets job properties common to pipeline jobs which are responsible for being the root of a
+   * build tree. Downstream jobs should pull artifacts from these jobs.
+   * @param context The delegate from the top level of a MavenJob.
+   */
+  static def setPipelineBuildJobProperties(def context) {
+    context.properties {
+      githubProjectUrl('https://github.com/apache/beam/')
+    }
+
+    context.parameters {
+      stringParam(
+              'commit',
+              'master',
+              'Commit id or refname (e.g. origin/pr/9/head) you want to build.')
+    }
+
+    // Source code management.
+    setBeamSCM(context, 'beam')
+  }
+
+  /**
+   * Sets job parameters common to pipeline jobs which are downstream of a job which builds
+   * artifacts for them.
```

Comment:
<!--new_thread; commit:13434ff7764e80f3ff96f6120b9100f537df0f36; resolved:0-->
This is pretty difficult to parse; "Sets common job parameters for (downstream jobs). Must be called from a job that produces artifacts for those jobs." I think is equivalent.

Reply:
Not quite right, but I gave rewriting this comment a shot -- let me know what you think.